### PR TITLE
Dl add backend

### LINKED ISSUE #10: Implement Backend Functionality: Generate Practice Exam Questions & Update on Submission
allow user to generate assembled_exam_questions for a practice exam as well as create that practice exam. Also allow for user to take practice exam as a form and update the practice exam.

### PR DESCRIPTION
**Pull request by DavidVLe1**
closes #10

### CODE REVIEW

File: app/controllers/exams_controller.rb [hunk 1]
```diff
@@ -67,4 +88,23 @@ def set_exam
...
+        selected_choices.each do |choice|
+          AssembledExamQuestion.create(
+            practice_exam: practice_exam,
+            question: question,
+            question_choice: choice
+          )
+        end
+      end
+    end
 end
```
Comment (phamdvincent) [inline app/controllers/exams_controller.rb:127]: With a method this long, you can split it into multiple methods.

**Review by phamdvincent — APPROVED** [review]
Good overall! Everything seems to be working, just make sure to clean your code so the methods can be more concise and readable.

**Review by WCW789 — APPROVED** [review]
Nice job on your final project so far! My questions have already been discussed in person, and it sounds like you will be removing most of the links in the "Resources" dropdown and revising the practice exams page to only show the current user's exams. The score shown is actually a percentage, and you have already planned to add a % at the end

File: app/controllers/exams_controller.rb (hunk 1 above)
Comment (DavidVLe1) [reply app/controllers/exams_controller.rb:127]: Thanks for the advice, I split it!

**Pull request merged**